internal/controllers/clusterextension_controller_test.go

```diff
@@ -1445,3 +1447,48 @@ var testBundleList = []*catalogmetadata.Bundle{
 		InChannels:  []*catalogmetadata.Channel{&prometheusBetaChannel},
 	},
 }
+
+func TestClusterExtensionErrorGettingBundles(t *testing.T) {
+	ctx := context.Background()
+	fakeBundleProvider := testutil.NewFakeCatalogClientWithError(errors.New("fake-test-error"))
+	cl, reconciler := newClientAndReconciler(t, nil)
+	reconciler.BundleProvider = &fakeBundleProvider
+	extKey := types.NamespacedName{Name: fmt.Sprintf("cluster-extension-test-%s", rand.String(8))}
+
+	t.Log("Creating a test cluster extension object")
+	clusterExtension := &ocv1alpha1.ClusterExtension{
+		ObjectMeta: metav1.ObjectMeta{Name: extKey.Name},
+		Spec: ocv1alpha1.ClusterExtensionSpec{
+			PackageName:      "prometheus",
+			InstallNamespace: "default",
+		},
+	}
+	err := cl.Create(ctx, clusterExtension)
+	defer func() {
+		require.NoError(t, cl.Delete(ctx, &ocv1alpha1.ClusterExtension{
+			ObjectMeta: metav1.ObjectMeta{Name: extKey.Name},
+		}))
+	}()
```

Comment:
Should this `defer` be _after_ the test for success (below)?

Reply:
@tmshort as far as I understand `require.NoError` exists immediately. I think if we `defer` after it - there is a chance that we will not clean up.

On the other hand the chance of `Create` returning non-nil error and creating the resources are very slim (Can It happen at all? E.g. transport error after creation?)

Reply:
Updated